src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests.cs

```diff
@@ -8083,6 +8323,41 @@ public class Inner
             Diagnostic(ErrorCode.ERR_AttributeCantBeGeneric, "Inner").WithArguments("Outer<int>.Inner"));
         }
 
+        [Fact]
+        public void AliasedGenericAttributeType_NestedWithGenericAttributeFeature()
+        {
+            var source = @"
+using InnerAlias = Outer<int>.Inner;
+using OuterAlias = Outer<int>;
+
+[InnerAlias]
+class Test
+{
+    [OuterAlias.Inner]
+    static void Main()
+    {
+    }
+}
+
+public class Outer<T>
+{
+    // Not a subtype of Attribute, since that wouldn't compile.
+    public class Inner 
```

Comment:
> // Not a subtype of Attribute, since that wouldn't compile.

Wait what?! But this is one of my use-cases. 💢

Reply:
Could you add a test for that scenario?
```
public class Outer<T>
{
    public class Inner : System.Attribute // error CS0698: A generic type cannot derive from 'Attribute' because it is an attribute class
    {
    }
}
```

If the reason we disallowed attribute types nested inside generic types was that you could not use the resulting generic, it feels like we should lift this restriction (conditional on the feature/LangVer)

---
In reply to: [339421395](https://github.com/dotnet/roslyn/pull/26337#discussion_r339421395) [](ancestors = 339421395)